app/models/Channel.ts

```diff
@@ -40,18 +49,17 @@ export const ChannelModel = types
       const uniqueEvents = events.filter(
         (obj, index) => events.findIndex((item) => item.id === obj.id) === index,
       )
-      self.setProp("messages", uniqueEvents)
-    },
-    async fetchMeta(channel: ChannelManager) {
-      const result = await channel.getMeta(self.id, self.privkey)
+      self.setProp("loading", false)
+      self.messages = cast(uniqueEvents)
```

Comment:
why cast vs setProp.   isn't setProp needed for the state-tree to detect the change?

Reply:
ya, they almost same, but setProp underhood is a bit tricky due to ignore typescript
```js
setProp<K extends keyof SnapshotIn<T>, V extends SnapshotIn<T>[K]>(field: K, newValue: V) {
    // @ts-ignore - for some reason TS complains about this, but it still works fine
    mstInstance[field] = newValue
},
```
`cast` help us figure out that this assignment is valid, just for better type support, example
<img width="592" alt="Screenshot 2023-06-30 at 09 13 44" src="https://github.com/ArcadeLabsInc/arcade/assets/123083837/3c40f330-13e1-44d1-a65b-c29b0a444220">